src/sync/streaming/UpdateWorkers/SplitsUpdateWorker.ts

```diff
@@ -79,9 +79,24 @@ export function SplitsUpdateWorker(log: ILogger, splitsCache: ISplitsCacheSync,
     backoff.reset();
   }
 
+  function putWithPayload(payload: any) {
```

Comment:
There is no need of duplicating code here. You can add an optional `payload` parameter to `put`